testsuite/cluster-test/src/experiments/reboot_cluster.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use std::{fmt, time::Duration};
+
+use crate::{
+    cluster::Cluster,
+    experiments::{Context, Experiment, ExperimentParam},
+    instance,
+    instance::Instance,
+};
+use async_trait::async_trait;
+use futures::future::try_join_all;
+use libra_logger::info;
+use std::{
+    collections::HashSet,
+    fmt::{Error, Formatter},
+};
+use structopt::StructOpt;
+use tokio::time;
+
+#[derive(StructOpt, Debug)]
+pub struct RebootClusterParams {}
+
+pub struct RebootCluster {
+    instances: Vec<Instance>,
+}
+
+impl ExperimentParam for RebootClusterParams {
+    type E = RebootCluster;
+    fn build(self, cluster: &Cluster) -> Self::E {
+        Self::E {
+            instances: <&[instance::Instance]>::clone(&cluster.validator_instances()).to_vec(),
+        }
+    }
+}
+
+#[async_trait]
+impl Experiment for RebootCluster {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&self.instances)
+    }
+
+    async fn run(&mut self, _context: &mut Context<'_>) -> anyhow::Result<()> {
+        let futures: Vec<_> = self.instances.iter().map(Instance::stop).collect();
+        try_join_all(futures).await?;
+        for inst in &self.instances {
+            info!("Starting node {}", inst.peer_name());
+            inst.start(false).await?;
+            time::delay_for(Duration::from_secs(10)).await;
```

Comment:
We need to have wait for json rpc here, instead of delay

Reply:
is this designed for slow startup?
```
start nodes one by one, with small period of sleep(~10 s) between starting each node.
```

Reply:
Ok, got it. Makes sense